yt_dlp/extractor/tver.py

```diff
@@ -161,17 +176,212 @@ def _real_extract(self, url):
+                'url': smuggle_url(
+                    self.BRIGHTCOVE_URL_TEMPLATE % (p_id, r_id),
+                    {'geo_countries': ['JP']}),
+                'ie_key': 'BrightcoveNew',
+            }
+
+        ref_id = video_info['streaks']['videoRefID']
+        if not ref_id.startswith('ref:'):
+            ref_id = f'ref:{ref_id}'
+
+        return {
+            **self._extract_from_streaks_api(video_info['streaks']['projectID'], ref_id, {
+                'Origin': 'https://tver.jp',
+                'Referer': 'https://tver.jp/',
+            }),
+            **metadata,
+            'id': video_id,
+        }
+
+    def _format_broadcast_date(self, onair_label):
+        """
+        Extracts the broadcast date from the onair label
+
+        Truth to be said, we cannot be sure or guarantee that the broadcast date is correct
+        as TVer doesn't really have consistent date format for the broadcast date.
+        At best we can only assume the following:
+        - If there is only year, this mean the broadcast is old.
+        - If there is only month and day, this mean the broadcast is recent within the current year or the previous year.
+
+        :param onair_label: The onair label string
+        :return: A dictionary containing the formatted broadcast date or an empty dictionary if the date is not found
+
+        """
+        if not onair_label:
+            return {}
+
+        match = re.search(
+            pattern=r'(?:(?P<year>\d{4})年)|(?:(?P<month>\d{1,2})\D(?P<day>\d{1,2})\D)',
+            string=onair_label,
+        )
+
+        if not match:
+            return {}
+
+        data = {}
+
+        broadcast_date = match.groupdict()
+
+        if broadcast_date.get('year'):
+            data['release_year'] = int(broadcast_date['year'])
+
+        if broadcast_date.get('day') and broadcast_date.get('month'):
+            if 'release_year' in data:
+                year = data['release_year']
+            else:
+                year = dt.datetime.now().year
+                dt_ = dt.datetime.strptime(
+                    f"{year}-{broadcast_date['month']}-{broadcast_date['day']}",
+                    '%Y-%m-%d',
+                )
+                # if the date in the future, it means the broadcast date is in the previous year
+                # ref: https://github.com/yt-dlp/yt-dlp/pull/12282#issuecomment-2678132806
+                if dt_ > dt.datetime.now():
+                    year -= 1
+
+            data['release_timestamp'] = dt.datetime(
+                year=year,
+                month=int(broadcast_date['month']),
+                day=int(broadcast_date['day']),
+            ).timestamp()
+
+        return data
+
+    # XXX: Remove everything below and subclass TVerIE from StreaksBaseIE when #12679 is merged
+    _API_URL_TEMPLATE = 'https://{}.api.streaks.jp/v1/projects/{}/medias/{}{}'
+
+    def _extract_from_streaks_api(self, project_id, media_id, headers=None, query=None, ssai=False):
```

Comment:
This and everything below this is just duplicated from #12679. Feel free to test this, but any suggestions and review should be done in #12679 for this code

Reply:
I didn't import  the `_extract_from_streaks_api` changes as requested in discord